### PR TITLE
FIX: Fix weekdays incorrect computation on midnight events in TimeZones with offset

### PR DESCRIPTION
**Pull request by KyMback**

### CODE REVIEW

File: src/Webinex.Calendar.Tests/RepeatEventCalculatorTests/RepeatEventCalculatorTests_Weekday.cs [hunk 1]
```diff
@@ -68,7 +68,7 @@ public void WhenOverStartOfDst_ShouldMatch()
     {
         new RepeatEventCalculatorScenario()
             .WithRange(DateTimeOffset.Parse("2023-10-22T00:00:00+000"), DateTimeOffset.Parse("2023-11-05T02:30:00+000"))
-            .WithWeekdayMatch("01:30", "1:00", "Europe/London", Weekday.Sunday)
+            .WithWeekdayMatch("01:30", "1:00", "Europe/London", 1, DateTimeOffset.Parse("2023-10-22T00:30:00+000"), null, Weekday.Sunday)
```
Comment (s-skalaban-itra) [inline src/Webinex.Calendar.Tests/RepeatEventCalculatorTests/RepeatEventCalculatorTests_Weekday.cs:71]: please, use named arguments for numbers and null

File: src/Webinex.Calendar/Repeats/Calculators/WeekdayRepeatEventCalculator.cs [hunk 2]
```diff
@@ -81,7 +84,6 @@ private Period Map(RecurrentEvent @event, Occurrence x)
 
     private LocalDateTime Tz(DateTimeOffset value, string tz)
     {
-        var localDateTime = value.ToUtc().DateTime.ToLocalDateTime();
-        return localDateTime.InZoneLeniently(DateTimeZoneProviders.Tzdb[tz]).LocalDateTime;
+        return value.ToInstant().InZone(DateTimeZoneProviders.Tzdb[tz]).LocalDateTime;
```
Comment (webinex) [inline src/Webinex.Calendar/Repeats/Calculators/WeekdayRepeatEventCalculator.cs:87]: Actually, utc can be converted to zone without a problems with ambigues time and double time. So it's safe

File: src/Webinex.Calendar.Tests/RepeatEventCalculatorTests/RepeatEventCalculatorTests_Weekday.cs [hunk 3]
```diff
@@ -86,7 +86,7 @@ public void WhenOverEndOfDst_ShouldMatch()
     {
         new RepeatEventCalculatorScenario()
             .WithRange(DateTimeOffset.Parse("2024-03-24T01:30:00+000"), DateTimeOffset.Parse("2024-04-07T02:30:00+000"))
-            .WithWeekdayMatch("01:30", "1:00", "Europe/London", Weekday.Sunday)
+            .WithWeekdayMatch("01:30", "1:00", "Europe/London", 1, DateTimeOffset.Parse("2024-03-24T01:30:00+000"), null, Weekday.Sunday)
```
Comment (webinex) [inline src/Webinex.Calendar.Tests/RepeatEventCalculatorTests/RepeatEventCalculatorTests_Weekday.cs]: please, use named arguments for nulls and numbers

**Review by webinex — APPROVED** [review]

File: src/Webinex.Calendar/Repeats/Calculators/WeekdayRepeatEventCalculator.cs (hunk 2 above)
Comment (KyMback) [reply src/Webinex.Calendar/Repeats/Calculators/WeekdayRepeatEventCalculator.cs:87]: yes, any instant/global time could be converted to local time, but not all local times are unambiguous/exist

**Pull request merged**